lib/clAdapters/activeSync/calendar/index.ts

```diff
@@ -0,0 +1,310 @@
+        device: {
+          id: '0000000',
+          type: 'iPhone',
+          model: 'iPhone Simulator',
+          operatingSystem: 'iPhone OS8.1'
+        }
+      };
+
+      const myCalClient = asclient(options);
+
+      await myCalClient.provision();
+
+      const folderSync = (await myCalClient.folderSync()).body.FolderSync;
+      // const folderSyncKey = folderSync.SyncKey;
+
+      // console.log('syncKey', folderSyncKey );
+
+      const calFolder = _.find(options.folders, (folder: any) => {
+        return folder.Type[0] === '8'; // Calendar type
+      });
+
+      const serverId = calFolder.ServerId[0];
+
+      // options.folderSyncKey = '1010372622';
+      await myCalClient.enableCalendarSync();
+      await myCalClient.sync();
+
+      const contents = myCalClient.contents;
+      const folders: any[] = contents[serverId];
+
+      const startDate = moment(filterStartDate);
+      const endDate = moment(filterEndDate);
+
+      const events = _.compact(_.map(folders, (folder: any) => {
+        const event = folder.ApplicationData[0];
+        const startTime = moment(event.StartTime[0]);
+        const endTime = moment(event.EndTime[0]);
+
+        if (startTime.isBefore(startDate) || endTime.isAfter(endDate) ) {
+          return null;
+        }
+
+        return event;
+      }));
+
+      const mappedEvents = _.map(events || [], (originalEvent: any) => {
+        const mappedEvent: any = {};
+        // console.log('event', JSON.stringify(originalEvent, null, 2));
+
+        // change to desired names
+        _.each(fieldNameMap, (have: string, want: string) => {
+          const val = _.get(originalEvent, have);
+          const mapped = val && val.length ? val[0] : val;
+
+          if (mapped !== undefined) {
+            mappedEvent[want] = /^dateTime/.test(want) ? moment(mapped).toDate() : mapped;
+          }
+        });
+
+        if (mappedEvent.responseStatus && mappedEvent.responseStatus.Response) {
+          mappedEvent.responseStatus = mappedEvent.responseStatus.Response;
+        }
+
+        const attendees = originalEvent[fieldNameMap['attendees']];
+
+        if (attendees && attendees.length) {
+          const attendeePeople = attendees[0].Attendee;
+
+          mappedEvent['attendees'] = attendeePeople
+            .map((attendee: any) => {
+              let acceptedStatus: string = _.get(attendee, 'AttendeeStatus[0]') || '0';
+
+              if (acceptedStatus === ACCEPTED_STATUS) {
+                acceptedStatus = 'Accepted';
+              }
+
+              return {
+                address:  _.get(attendee, 'Email[0]'),
+                name:     _.get(attendee, 'Name[0]'),
+                response: acceptedStatus
+              };
+            });
+        } else {
+          mappedEvent['attendees'] = [];
+        }
+
+        // Make it the same as the eventId for now, since it does not look like this value available
+        mappedEvent.iCalUId = mappedEvent.eventId;
```

Comment:
I set it here because TmCalendarAdapterBatchHandler is using this as a hashkey for the events, and will fail if it is not defined.